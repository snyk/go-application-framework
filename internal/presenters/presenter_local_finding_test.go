package presenters_test

import (
	"bytes"
	"encoding/json"
	"io"
	"os"
	"testing"
	"text/template"

	"github.com/charmbracelet/lipgloss"
	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/muesli/termenv"
	"github.com/snyk/code-client-go/sarif"
	"github.com/stretchr/testify/assert"
	"github.com/stretchr/testify/require"

	"github.com/snyk/go-application-framework/internal/presenters"
	sarif_utils "github.com/snyk/go-application-framework/internal/utils/sarif"
	"github.com/snyk/go-application-framework/pkg/configuration"
	localworkflows "github.com/snyk/go-application-framework/pkg/local_workflows"
	"github.com/snyk/go-application-framework/pkg/local_workflows/local_models"
)

func sarifToLocalFinding(t *testing.T, filename string) (localFinding *local_models.LocalFinding, err error) {
	t.Helper()
	jsonFile, err := os.Open("./" + filename)
	if err != nil {
		t.Errorf("Failed to load json")
	}

	defer func(jsonFile *os.File) {
		jsonErr := jsonFile.Close()
		assert.NoError(t, jsonErr)
	}(jsonFile)
	sarifBytes, err := io.ReadAll(jsonFile)
	assert.NoError(t, err)

	// Read sarif file again for summary
	var sarifDoc sarif.SarifDocument

	err = json.Unmarshal(sarifBytes, &sarifDoc)
	assert.NoError(t, err)

	summaryData := sarif_utils.CreateCodeSummary(&sarifDoc, "/path/to/project")
	summaryBytes, err := json.Marshal(summaryData)
	assert.NoError(t, err)

	tmp, err := localworkflows.TransformToLocalFindingModel(sarifBytes, summaryBytes)
	return &tmp, err
}

func TestPresenterLocalFinding_NoIssues(t *testing.T) {
	fd, err := os.Open("testdata/local-findings-empty.json")
	require.NoError(t, err)

	var localFindingDoc *local_models.LocalFinding
	err = json.NewDecoder(fd).Decode(&localFindingDoc)
	require.NoError(t, err)

	scannedPath := "/path/to/project"
	lipgloss.SetColorProfile(termenv.Ascii)
	config := configuration.NewInMemory()
	config.Set(configuration.ORGANIZATION_SLUG, "test-org")
	writer := new(bytes.Buffer)

	p := presenters.NewLocalFindingsRenderer(
		[]*local_models.LocalFinding{localFindingDoc},
		config,
		writer,
	)

	err = p.RenderTemplate(presenters.DefaultTemplateFiles, presenters.DefaultMimeType)
	result := writer.String()

	require.NoError(t, err)
	assert.Contains(t, result, "Testing "+scannedPath)
	assert.NotContains(t, result, "Ignored issues")
	snaps.MatchSnapshot(t, result)
}

func TestPresenterLocalFinding_LowIssues(t *testing.T) {
	// Convert our sarif into localfindings
	input, err := sarifToLocalFinding(t, "testdata/3-low-issues.json")
	require.NoError(t, err)

	lipgloss.SetColorProfile(termenv.Ascii)
	config := configuration.NewInMemory()
	config.Set(configuration.ORGANIZATION_SLUG, "test-org")
	writer := new(bytes.Buffer)

	p := presenters.NewLocalFindingsRenderer(
		[]*local_models.LocalFinding{input},
		config,
		writer,
	)

	err = p.RenderTemplate(presenters.DefaultTemplateFiles, presenters.DefaultMimeType)
	result := writer.String()

	require.NoError(t, err)
	snaps.MatchSnapshot(t, result)
}

func TestPresenterLocalFinding_MediumHighIssues(t *testing.T) {
	input, err := sarifToLocalFinding(t, "testdata/4-high-5-medium.json")
	require.Nil(t, err)

	lipgloss.SetColorProfile(termenv.Ascii)
	config := configuration.NewInMemory()
	config.Set(configuration.ORGANIZATION_SLUG, "test-org")
	writer := new(bytes.Buffer)

	p := presenters.NewLocalFindingsRenderer(
		[]*local_models.LocalFinding{input},
		config,
		writer,
	)

	err = p.RenderTemplate(presenters.DefaultTemplateFiles, presenters.DefaultMimeType)
	result := writer.String()

	require.Nil(t, err)

	require.Contains(t, result, "[ 0 HIGH  0 MEDIUM  0 LOW ]")
	require.Contains(t, result, "[ 4 HIGH  5 MEDIUM  0 LOW ]")
	snaps.MatchSnapshot(t, result)
}

func TestPresenterLocalFinding_MediumHighIssuesWithColor(t *testing.T) {
	input, err := sarifToLocalFinding(t, "testdata/4-high-5-medium.json")
	require.Nil(t, err)

	lipgloss.SetColorProfile(termenv.TrueColor)
	config := configuration.NewInMemory()
	config.Set(configuration.ORGANIZATION_SLUG, "test-org")
	writer := new(bytes.Buffer)

	p := presenters.NewLocalFindingsRenderer(
		[]*local_models.LocalFinding{input},
		config,
		writer,
	)

	err = p.RenderTemplate(presenters.DefaultTemplateFiles, presenters.DefaultMimeType)
	result := writer.String()

	require.Nil(t, err)
	require.NotContains(t, result, "You are currently viewing results with --severity-threshold=high applied")

	snaps.MatchSnapshot(t, result)
}

func TestPresenterLocalFinding_MediumHighIssuesWithColorLight(t *testing.T) {
	input, err := sarifToLocalFinding(t, "testdata/4-high-5-medium.json")
	require.Nil(t, err)

	lipgloss.SetColorProfile(termenv.TrueColor)
	lipgloss.SetHasDarkBackground(false)
	config := configuration.NewInMemory()
	config.Set(configuration.ORGANIZATION_SLUG, "test-org")
	writer := new(bytes.Buffer)

	p := presenters.NewLocalFindingsRenderer(
		[]*local_models.LocalFinding{input},
		config,
		writer,
	)

	err = p.RenderTemplate(presenters.DefaultTemplateFiles, presenters.DefaultMimeType)
	result := writer.String()

	require.Nil(t, err)
	require.NotContains(t, result, "You are currently viewing results with --severity-threshold=high applied")

	snaps.MatchSnapshot(t, result)
}

func TestPresenterLocalFinding_DefaultHideIgnored(t *testing.T) {
	input, err := sarifToLocalFinding(t, "testdata/with-ignores.json")
	require.Nil(t, err)

	lipgloss.SetColorProfile(termenv.Ascii)
	config := configuration.NewInMemory()
	config.Set(configuration.ORGANIZATION_SLUG, "test-org")
	config.Set(configuration.FLAG_INCLUDE_IGNORES, false)
	writer := new(bytes.Buffer)

	p := presenters.NewLocalFindingsRenderer(
		[]*local_models.LocalFinding{input},
		config,
		writer,
	)

	err = p.RenderTemplate(presenters.DefaultTemplateFiles, presenters.DefaultMimeType)
	result := writer.String()

	require.Nil(t, err)
	require.NotContains(t, result, "src/main.ts, line 58")
	require.NotContains(t, result, "Ignored Issues")
}

func TestPresenterLocalFinding_IncludeIgnored(t *testing.T) {
	input, err := sarifToLocalFinding(t, "testdata/with-ignores.json")
	require.Nil(t, err)

	lipgloss.SetColorProfile(termenv.Ascii)
	config := configuration.NewInMemory()
	config.Set(configuration.ORGANIZATION_SLUG, "test-org")
	config.Set(configuration.FLAG_INCLUDE_IGNORES, true)
	writer := new(bytes.Buffer)

	p := presenters.NewLocalFindingsRenderer(
		[]*local_models.LocalFinding{input},
		config,
		writer,
	)

	err = p.RenderTemplate(presenters.DefaultTemplateFiles, presenters.DefaultMimeType)
	result := writer.String()

	require.Nil(t, err)

	require.Contains(t, result, "Ignored Issues")
	require.Contains(t, result, "[ IGNORED ] [MEDIUM]")
	require.Contains(t, result, "src/main.ts, line 58")
	require.Contains(t, result, "Ignores are currently managed in the Snyk Web UI.")
	require.NotContains(t, result, "Empty ignore issues state")
	require.NotContains(t, result, "To view ignored and open issues, use the --include-ignores option.pre")

	snaps.MatchSnapshot(t, result)
}

func TestPresenterLocalFinding_IncludeIgnoredEmpty(t *testing.T) {
	input, err := sarifToLocalFinding(t, "testdata/3-low-issues.json")
	require.Nil(t, err)

	lipgloss.SetColorProfile(termenv.Ascii)

	config := configuration.NewInMemory()
	config.Set(configuration.ORGANIZATION_SLUG, "test-org")
	config.Set(configuration.FLAG_INCLUDE_IGNORES, true)
	writer := new(bytes.Buffer)

	p := presenters.NewLocalFindingsRenderer(
		[]*local_models.LocalFinding{input},
		config,
		writer,
	)

	err = p.RenderTemplate(presenters.DefaultTemplateFiles, presenters.DefaultMimeType)
	result := writer.String()

	require.Nil(t, err)
	require.NotContains(t, result, "[ IGNORED ]")
	require.Contains(t, result, "There are no ignored issues")

	snaps.MatchSnapshot(t, result)
}

func TestPresenterLocalFinding_CustomTemplateFiles(t *testing.T) {
	input, err := sarifToLocalFinding(t, "testdata/3-low-issues.json")
	require.Nil(t, err)
	config := configuration.NewInMemory()
	config.Set(configuration.ORGANIZATION, "org-id1267361872673627")
	config.Set(configuration.ORGANIZATION_SLUG, "test-org")
	writer := new(bytes.Buffer)
	p := presenters.NewLocalFindingsRenderer(
		[]*local_models.LocalFinding{input},
		config,
		writer,
	)

	t.Run("no template files", func(t *testing.T) {
		err = p.RenderTemplate([]string{}, presenters.DefaultMimeType)
		assert.Error(t, err)
	})

	t.Run("not existing template file", func(t *testing.T) {
		err = p.RenderTemplate([]string{"notexstingfile.tmpl"}, presenters.DefaultMimeType)
		assert.Error(t, err)
	})

	t.Run("unknown mimetype", func(t *testing.T) {
		err = p.RenderTemplate([]string{"testdata/custom_template_html.tmpl"}, "application/me")
		assert.NoError(t, err)
	})

	t.Run("custom template file missing main", func(t *testing.T) {
		err = p.RenderTemplate([]string{"testdata/custom_template_no_main.tmpl"}, "application/json")
		assert.Error(t, err)
	})

	t.Run("custom template file (with whitespaces)", func(t *testing.T) {
		writer.Reset()

		err = p.RenderTemplate([]string{"testdata/custom_template.tmpl"}, "application/json")
		assert.NoError(t, err)

		expected := `{
    "findings" :
    [
        "Use of Hardcoded Credentials",
        "Use of Hardcoded Credentials",
        "Use of Password Hash With Insufficient Computational Effort"
    ]
}`

		actual := writer.String()
		assert.Equal(t, expected, actual)
	})

	t.Run("custom template file (without whitespaces)", func(t *testing.T) {
		writer.Reset()
		config.Set(presenters.CONFIG_JSON_STRIP_WHITESPACES, true)

		err = p.RenderTemplate([]string{"testdata/custom_template.tmpl"}, "application/json")
		assert.NoError(t, err)

		expected := `{"findings" :["Use of Hardcoded Credentials","Use of Hardcoded Credentials","Use of Password Hash With Insufficient Computational Effort"]}`

		actual := writer.String()
		assert.Equal(t, expected, actual)
	})
}

func TestPresenterLocalFinding_RegisterMimeType(t *testing.T) {
	config := configuration.NewInMemory()
	writer := new(bytes.Buffer)
	p := presenters.NewLocalFindingsRenderer(nil, config, writer)

	t.Run("try registering existing type", func(t *testing.T) {
		err := p.RegisterMimeType(presenters.DefaultMimeType, func() (*template.Template, template.FuncMap, error) {
			return nil, nil, nil
		})
		assert.Error(t, err)
	})

	t.Run("try registering a new type", func(t *testing.T) {
		err := p.RegisterMimeType("mymime", func() (*template.Template, template.FuncMap, error) {
			return nil, nil, nil
		})
		assert.NoError(t, err)
	})
}

func TestJsonWriter(t *testing.T) {
	t.Run("strip whitespaces while writing", func(t *testing.T) {
		buffer := &bytes.Buffer{}
		writerUnderTest := presenters.NewJsonWriter(buffer, true)

		input := []byte(`{
	"name": "myName",
    "address": "myAddr"
}`)

		expected := `{"name": "myName","address": "myAddr"}`

		bytesWritten, err := writerUnderTest.Write(input)
		assert.NoError(t, err)
		assert.Equal(t, len(input), bytesWritten)
		assert.Equal(t, expected, buffer.String())
	})

	t.Run("Don't strip whitespaces while writing", func(t *testing.T) {
		buffer := &bytes.Buffer{}
		writerUnderTest := presenters.NewJsonWriter(buffer, false)

		input := []byte(`{
	"name": "myName",
    "address": "myAddr"
}`)

		bytesWritten, err := writerUnderTest.Write(input)
		assert.NoError(t, err)
		assert.Equal(t, len(input), bytesWritten)
		assert.Equal(t, input, buffer.Bytes())
	})
}
