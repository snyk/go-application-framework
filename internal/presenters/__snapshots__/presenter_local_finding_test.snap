
[TestPresenterLocalFinding_with_Issues - 1]

Testing path/to/project ...

Open Issues

[91mâœ— [HIGH][0m [1mServer-Side Request Forgery (SSRF)[0m
   Path: routes/profileImageUrlUpload.ts, line 22
   Info: Unsanitized input from the HTTP request body flows into request.get, where it is used as an URL to perform a request. This may result in a Server-Side Request Forgery vulnerability.


â•­â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â•®
â”‚ Test Summary                                                   â”‚
â”‚                                                                â”‚
â”‚   Organization:                                                â”‚
â”‚   Test type:         Static code analysis                      â”‚
â”‚   Project path:      path/to/project                           â”‚
â”‚                                                                â”‚
â”‚   Total issues:   18                                           â”‚
â”‚   Ignored issues: [1m3[0m [ 0 CRITICAL [91m 2 HIGH [0m[93m 1 MEDIUM [0m 0 LOW ]    â”‚
â”‚   Open issues:    [1m5[0m [ 1 CRITICAL [91m 3 HIGH [0m[93m 1 MEDIUM [0m 0 LOW ]    â”‚
â•°â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â•¯

ðŸ’¡ Tip

    To view ignored issues, use the --include-ignores option.


---

[TestBoxStyle - 1]
â•­â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â•®
â”‚ [91mâœ— [HIGH][0m [1mCleartext Transmission of Sensitive Information[0m                                                                                                                                                                                                                        â”‚
â”‚    Path: node_modules/qs/support/expresso/test/http.test.js, line 1                                                                                                                                                                                                             â”‚
â”‚    Info: http.createServer uses HTTP which is an insecure protocol and should not be used in code due to cleartext transmission of information. Data in cleartext in a communication channel can be sniffed by unauthorized actors. Consider using the https module instead.    â”‚
â”‚                                                                                                                                                                                                                                                                                 â”‚
â”‚                                                                                                                                                                                                                                                                                 â”‚
â•°â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â•¯
---

[TestFindingComponent - 1]
[91mâœ— [HIGH][0m [1mCleartext Transmission of Sensitive Information[0m
   Path: node_modules/qs/support/expresso/test/http.test.js, line 1
   Info: http.createServer uses HTTP which is an insecure protocol and should not be used in code due to cleartext transmission of information. Data in cleartext in a communication channel can be sniffed by unauthorized actors. Consider using the https module instead.


---

[TestPresenterLocalFinding_with_severityFilter - 1]

Testing path/to/project ...

Open Issues

[91mâœ— [HIGH][0m [1mServer-Side Request Forgery (SSRF)[0m
   Path: routes/profileImageUrlUpload.ts, line 22
   Info: Unsanitized input from the HTTP request body flows into request.get, where it is used as an URL to perform a request. This may result in a Server-Side Request Forgery vulnerability.


ðŸ’¡ Tip

    You are currently viewing results with --severity-threshold applied.
    To view all issues, remove the --severity-threshold flag

â•­â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â•®
â”‚ Test Summary                                  â”‚
â”‚                                               â”‚
â”‚   Organization:                               â”‚
â”‚   Test type:         Static code analysis     â”‚
â”‚   Project path:      path/to/project          â”‚
â”‚                                               â”‚
â”‚   Total issues:   11                          â”‚
â”‚   Ignored issues: [1m2[0m [ 0 CRITICAL [91m 2 HIGH [0m]    â”‚
â”‚   Open issues:    [1m4[0m [ 1 CRITICAL [91m 3 HIGH [0m]    â”‚
â•°â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â•¯

ðŸ’¡ Tip

    To view ignored issues, use the --include-ignores option.


---

[TestPresenterLocalFinding_with_ignored - 1]

Testing path/to/project ...

Open Issues

âœ— [HIGH] Server-Side Request Forgery (SSRF)
   Path: routes/profileImageUrlUpload.ts, line 22
   Info: Unsanitized input from the HTTP request body flows into request.get, where it is used as an URL to perform a request. This may result in a Server-Side Request Forgery vulnerability.


â•­â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â•®
â”‚ Test Summary                                                   â”‚
â”‚                                                                â”‚
â”‚   Organization:                                                â”‚
â”‚   Test type:         Static code analysis                      â”‚
â”‚   Project path:      path/to/project                           â”‚
â”‚                                                                â”‚
â”‚   Total issues:   18                                           â”‚
â”‚   Ignored issues: 3 [ 0 CRITICAL  2 HIGH  1 MEDIUM  0 LOW ]    â”‚
â”‚   Open issues:    5 [ 1 CRITICAL  3 HIGH  1 MEDIUM  0 LOW ]    â”‚
â•°â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â•¯
---

[TestPresenterLocalFinding_with_includeIgnores - 1]

Testing path/to/project ...

Open Issues

[91mâœ— [HIGH][0m [1mServer-Side Request Forgery (SSRF)[0m
   Path: routes/profileImageUrlUpload.ts, line 22
   Info: Unsanitized input from the HTTP request body flows into request.get, where it is used as an URL to perform a request. This may result in a Server-Side Request Forgery vulnerability.


â•­â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â•®
â”‚ Test Summary                                                   â”‚
â”‚                                                                â”‚
â”‚   Organization:                                                â”‚
â”‚   Test type:         Static code analysis                      â”‚
â”‚   Project path:      path/to/project                           â”‚
â”‚                                                                â”‚
â”‚   Total issues:   18                                           â”‚
â”‚   Ignored issues: [1m3[0m [ 0 CRITICAL [91m 2 HIGH [0m[93m 1 MEDIUM [0m 0 LOW ]    â”‚
â”‚   Open issues:    [1m5[0m [ 1 CRITICAL [91m 3 HIGH [0m[93m 1 MEDIUM [0m 0 LOW ]    â”‚
â•°â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â•¯
---
